CAE_datacleaning/g032_cae_calib.ipynb

```diff
@@ -0,0 +1,556 @@
+    "# have to process raw file using raw_to_dsp.py from pygama first, then save waveforms and other parameters into npy file\n",
+    "\n",
+    "#load data print dimensions of data set\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "waveforms = np.load('g032_waveforms.npy')\n",
+    "psd_sum10 = np.load('g032_psd_sum10.npy')\n",
+    "bl_slope  = np.load('g032_bl_slope.npy')\n",
+    "tp_0 = np.load('g032_tp_0.npy')\n",
+    "tp_50 = np.load('g032_tp_50.npy')\n",
+    "\n",
+    "train_data = waveforms[0:66200,:]\n",
+    "test_data = waveforms[66200:79400,:] #lose 34 observations for validation\n",
+    "test_psd10 = psd_sum10[66200:79400]\n",
+    "test_bl_slope = bl_slope[66200:79400]\n",
+    "test_tp_0= tp_0[66200:79400]\n",
+    "test_tp_50 = tp_50[66200:79400]\n",
+    "\n",
+    "print(waveforms.shape)\n",
+    "print(train_data.shape)\n",
+    "print(test_data.shape)\n",
+    "print(test_psd10.shape)\n",
+    "print(test_bl_slope.shape)\n",
+    "print(test_tp_0.shape)\n",
+    "print(test_tp_0.shape)\n",
+    "\n",
+    "plt.plot(np.arange(len(waveforms[374,:])), waveforms[374,:]) #anomalous sample wf \n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Convert data to tensors\n",
+    "train_data, test_data = map(torch.tensor, (train_data, test_data))\n",
+    "\n",
+    "# batch size \n",
+    "bs = 50 \n",
+    "\n",
+    "#train data loader\n",
+    "train_ds = TensorDataset(train_data)\n",
+    "train_dl = DataLoader(train_ds, batch_size=bs) #removed batch shuffling here\n",
+    "\n",
+    "#test data loader\n",
+    "valid_ds = TensorDataset(test_data)\n",
+    "valid_dl = DataLoader(valid_ds, batch_size=bs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Send data to GPU\n",
+    "def preprocess(x):\n",
+    "    return x.to(DEVICE)\n",
+    "\n",
+    "class WrappedDataLoader:\n",
+    "    def __init__(self, dl, func):\n",
+    "        self.dl = dl\n",
+    "        self.func = func\n",
+    "\n",
+    "    def __len__(self):\n",
+    "        return len(self.dl)\n",
+    "\n",
+    "    def __iter__(self):\n",
+    "        batches = iter(self.dl)\n",
+    "        for b in batches:\n",
+    "            yield (self.func(*b))\n",
+    "            \n",
+    "train_dl = WrappedDataLoader(train_dl, preprocess)\n",
+    "valid_dl = WrappedDataLoader(valid_dl,preprocess)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Define Convolutional Autoencoder\n",
+    "\n",
+    "class waveform_cae_ralph(nn.Module):\n",
+    "    def __init__(self):\n",
+    "        super(waveform_cae_ralph, self).__init__()\n",
+    "        \n",
+    "        self.dropout = nn.Dropout(p=0.1)\n",
```

Comment:
In the future you will want to have many dropout instances created (i.e. self.do1 = nn.Dropout(0.1), self.do2 = nn.Dropout(0.1)......) instead of using just one for all dropout operations.